src/sdam/topology_description.ts

```diff
@@ -362,25 +362,29 @@ function topologyTypeForServerType(serverType: ServerType): TopologyType {
   }
 }
 
-// TODO: improve these docs when ObjectId is properly typed
-function compareObjectId(oid1: Document, oid2: Document): number {
+function compareObjectId(oid1?: ObjectId, oid2?: ObjectId): 0 | 1 | -1 {
```

Comment:
Just my feeling here that `compareObjectId` and `compareNumber` might better be placed in the utils module as they are pretty generic.

Reply:
`compareObjectId` was moved to utils in #3348, rebased.